src/Router.tsx

```diff
@@ -1,16 +1,19 @@
-import React from 'react';
+import React, { lazy, Suspense } from 'react';
 import { BrowserRouter, Route, Routes } from 'react-router-dom';
 
-import Signin from './components/signin/signin';
 import { APP_ROUTES } from './enums';
-import Home from './pages/home/home';
+
+const Signin = lazy(() => import('./components/signin/signin'));
+const Home = lazy(() => import('./pages/home/home'));
 
 const Router: React.FC = () => (
   <BrowserRouter>
-    <Routes>
-      <Route path={APP_ROUTES.SIGNIN} element={<Signin />} />
-      <Route path={APP_ROUTES.HOME} element={<Home />} />
-    </Routes>
+    <Suspense fallback={<div>Download...</div>}>
+      <Routes>
+        <Route path={APP_ROUTES.SIGNIN} element={<Signin />} />
```

Comment:
is is better to suspense each route separately.

You can create such high-ordero-component for that:
```
interface DtWithSuspenseProps {
  Component?: LazyExoticComponent<ComponentType>;
  loader?: ReactElement;
}

const DEFAULT_LOADER = <DtSkeleton variant='rectangular' />;

/**
 * This HOC intended to wrap components that you want to lazy load {@link https://reactjs.org/docs/code-splitting.html}
 *
 * @param {React.LazyExoticComponent<React.FC>} Component Component that you want to exclude from bundle and lazy load on demand
 * @param {React.ReactElement} loader Component that will be displayed till component load
 *
 * @example const componentToBeLazyLoaded = withSuspense(lazy(() => import(pathToComponent)));
 */
export const dtWithSuspense =
  (Component: LazyExoticComponent<FC>, loader: ReactElement | undefined = DEFAULT_LOADER): FC<DtWithSuspenseProps> =>
  (props): ReactElement => {
    return (
      <Suspense fallback={loader}>
        <Component {...props} />
      </Suspense>
    );
  };
```